snapcraft/tests/fake_servers.py

```diff
@@ -45,14 +46,15 @@ class FakeSSORequestHandler(BaseHTTPRequestHandler):
 
     def do_POST(self):
         parsed_path = urllib.parse.urlparse(self.path)
-        oauth_path = self._API_PATH + 'tokens/oauth'
-        if parsed_path.path.startswith(oauth_path):
-            self._handle_oauth()
+        tokens_discharge_path = self._API_PATH + 'tokens/discharge'
```

Comment:
sure, why not. I just hate urljoin a little because when I miss the trailing / it does things I don't expect. But if the tests pass, all's good.